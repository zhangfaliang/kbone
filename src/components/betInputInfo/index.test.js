import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import { BetInfo, BetInput, BetModule } from './index';
import Keyboard from '../keyboard';

describe('BetModule', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(
      <BetModule>
        <BetInfo />
        <BetInput />
        <Keyboard />
      </BetModule>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('handleCloseKeyboard', () => {
    expect.assertions(1);
    const wrapper = mount(
      <BetModule maskClosable={true}>
        <BetInfo />
        <BetInput />
        <Keyboard />
      </BetModule>
    );
    const handleKeyboardChange = wrapper.prop('handleKeyboardChange');
    const click = jest.fn(handleKeyboardChange);

    wrapper.simulate('click');
    expect(click()).toBe('handleKeyboardChange');
  });

  it('shouldComponentUpdate --isShow', () => {
    expect.assertions(1);
    const wrapper = mount(<BetModule isShow={false} />);
    wrapper.setProps({
      isShow: true,
    });
    expect(true).toBe(wrapper.state('isShowBetModule'));
  });

  it('handelFocusInput', () => {
    expect.assertions(1);
    const wrapper = mount(
      <BetModule isShow={true}>
        <BetInfo />
        <BetInput />
        <Keyboard />
      </BetModule>
    );
    const focusInput = wrapper.prop('focusInput');
    const click = jest.fn(focusInput);

    wrapper
      .find('BetModule')
      .find('BetInput')
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .simulate('click');
    expect(click()).toBe('focusInput');
  });
  it('handleBet', () => {
    expect.assertions(2);
    const wrapper = mount(
      <BetModule allAmount="1000" fillAmount="100" isShow={true}>
        <BetInput />
      </BetModule>
    );
    const handleBet = wrapper.prop('handleBet');
    const betClick = jest.fn(handleBet);
    const handleCloseBetModule = wrapper.prop('handleCloseBetModule');
    const closeBetModule = jest.fn(handleCloseBetModule);
    wrapper
      .find('BetModule')
      .find('BetInput')
      .children('div')
      .at(0)
      .find('Button')
      .at(1)
      .simulate('click');
    expect(betClick()).toBe('handleBet');
    expect(closeBetModule()).toBe('handleCloseBetModule');
  });
  it('handleCloseBetModule', () => {
    expect.assertions(1);
    const wrapper = mount(
      <BetModule isShow={true}>
        <BetInfo />
        <BetInput />
        <Keyboard />
      </BetModule>
    );
    const handleCloseBetModule = wrapper.prop('handleCloseBetModule');
    const click = jest.fn(handleCloseBetModule);

    wrapper
      .find('BetModule')
      .find('BetInfo')
      .children('div')
      .at(0)
      .children('div')
      .at(1)
      .simulate('click');
    expect(click()).toBe('handleCloseBetModule');
  });

  it('handleFillIn', () => {
    expect.assertions(1);
    const wrapper = mount(
      <BetModule allAmount="1000" fillAmount="100" isShow={true}>
        <BetInput />
      </BetModule>
    );
    const handleCloseBetModule = wrapper.prop('handleCloseBetModule');
    const click = jest.fn(handleCloseBetModule);

    wrapper
      .find('BetModule')
      .find('BetInput')
      .children('div')
      .at(0)
      .find('Button')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleCloseBetModule');
  });
  it('handleAllIn', () => {
    expect.assertions(1);
    const wrapper = mount(
      <BetModule allAmount="100" fillAmount="1000" isShow={true}>
        <BetInput />
      </BetModule>
    );
    const handleCloseBetModule = wrapper.prop('handleCloseBetModule');
    const click = jest.fn(handleCloseBetModule);

    wrapper
      .find('BetModule')
      .find('BetInput')
      .children('div')
      .at(0)
      .find('Button')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleCloseBetModule');
  });
  it('handleAllIn .', () => {
    expect.assertions(1);
    const wrapper = mount(
      <BetModule allAmount="100.45" fillAmount="1000" isShow={true}>
        <BetInput />
      </BetModule>
    );
    const handleCloseBetModule = wrapper.prop('handleCloseBetModule');
    const click = jest.fn(handleCloseBetModule);

    wrapper
      .find('BetModule')
      .find('BetInput')
      .children('div')
      .at(0)
      .find('Button')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleCloseBetModule');
  });
});

describe('BetInput', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(<BetInput />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('handleFillIn', () => {
    expect.assertions(1);
    const wrapper = mount(<BetInput allAmount="1000" fillAmount="100" />);
    const handleFillIn = wrapper.prop('handleFillIn');
    const click = jest.fn(handleFillIn);
    wrapper.simulate('click');
    wrapper
      .children('div')
      .at(0)
      .children('div')
      .at(1)
      .simulate('click');
    wrapper
      .children('div')
      .at(0)
      .children('div')
      .at(1)
      .find('Button')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleFillIn');
  });

  it('handleAllIn', () => {
    expect.assertions(1);
    const wrapper = mount(<BetInput allAmount="10" fillAmount="100" />);
    const handleAllIn = wrapper.prop('handleAllIn');
    const click = jest.fn(handleAllIn);

    wrapper
      .children('div')
      .at(0)
      .find('Button')
      .at(0)
      .simulate('click');
    expect(click()).toBe('handleAllIn');
  });

  it('betInfo', () => {
    expect.assertions(1);
    const wrapper = mount(<BetInput allAmount="10" fillAmount="100" />);
    const handleBet = wrapper.prop('handleBet');
    const click = jest.fn(handleBet);

    wrapper
      .children('div')
      .at(0)
      .find('Button')
      .at(1)
      .simulate('click');
    expect(click()).toBe('handleBet');
  });

  it('handleInputChange', () => {
    expect.assertions(1);
    const wrapper = mount(<BetInput allAmount="10" fillAmount="100" />);
    const handleInputChange = wrapper.prop('handleInputChange');
    const click = jest.fn(handleInputChange);
    expect(click()).toBe('handleInputChange');
  });
  it('handleFocusInput', () => {
    expect.assertions(1);
    const wrapper = mount(<BetInput allAmount="100" fillAmount="10000" />);
    const focusInput = wrapper.prop('focusInput');
    const click = jest.fn(focusInput);
    wrapper
      .find('BetInput')
      .children('div')
      .at(0)
      .children('div')
      .at(0)
      .simulate('click');
    expect(click()).toBe('focusInput');
  });
});

describe('BetInfo', async () => {
  it('base', () => {
    expect.assertions(1);
    const wrapper = render(<BetInfo />);
    expect(toString(wrapper)).toMatchSnapshot();
  });

  it('handeldeleteclick', () => {
    expect.assertions(1);
    const wrapper = mount(<BetInfo />);
    wrapper.find('BetInfo').simulate('click');
    const handleDelete = wrapper.prop('handleDelete');
    const click = jest.fn(handleDelete);
    wrapper
      .find('BetInfo')
      .children('div')
      .at(0)
      .children('div')
      .at(1)
      .simulate('click');
    expect(click()).toBe('handleDelete');
  });
  it('shouldComponentUpdate sp', () => {
    expect.assertions(2);
    const wrapper = mount(
      <BetInfo matchId="567" playId="4564" defaultChangeSp={false} defaultChangeHandicap={false} />
    );
    wrapper.setProps({
      sp: 1.56,
      handicap: 8,
      matchId: 'werwe',
      playId: 'werwer',
      optionId: 'werwerwer',
    });
    const isChangeSp = wrapper.state('isChangeSp');
    const isChangeHandicap = wrapper.state('isChangeHandicap');
    expect(false).toBe(isChangeSp);
    expect(false).toBe(isChangeHandicap);
  });
  it('shouldComponentUpdate sp', () => {
    expect.assertions(2);
    const wrapper = mount(<BetInfo sp={1.3} handicap={4} />);
    wrapper.setProps({
      sp: 1.56,
      handicap: 8,
    });
    const isChangeSp = wrapper.state('isChangeSp');
    const isChangeHandicap = wrapper.state('isChangeHandicap');
    expect(true).toBe(isChangeSp);
    expect(true).toBe(isChangeHandicap);
  });
});
