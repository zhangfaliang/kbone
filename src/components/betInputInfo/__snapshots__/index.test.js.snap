// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BetInfo base 1`] = `"[object Undefined]"`;

exports[`BetInput base 1`] = `
<div
  class="default-input"
>
  <div
    class="inputWrap"
  >
    <div
      class="input"
    >
      <span
        class="betAmount amountDisabled"
      >
        0
      </span>
      <span
        class="amountSuffix"
      >
        EUR
      </span>
    </div>
    <div
      class="possibleWinWrap"
    >
      <span />
      <span>
        0 EUR
      </span>
    </div>
  </div>
  <a
    class="all-in-button"
    href="javascript:;"
  >
    <span />
  </a>
  <div>
    <a
      class="bet-button"
      href="javascript:;"
    >
      <span />
    </a>
  </div>
</div>
`;

exports[`BetModule base 1`] = `
<div
  class="default-bet-module"
>
  <div
    class="-betinfor"
  >
    <div
      class="left"
    >
      <div
        class="optionName"
      >
        <span
          class=""
        />
        <span
          class="sp"
        />
      </div>
      <div
        class="gamePlayName"
      />
      <div
        class="gameAgainst"
      />
    </div>
    <div
      class="right"
    >
      <span
        class="iconfont icon-delete"
      />
    </div>
  </div>
  <div
    class="default-input"
  >
    <div
      class="inputWrap"
    >
      <div
        class="input"
      >
        <span
          class="betAmount amountDisabled"
        >
          0
        </span>
        <span
          class="amountSuffix"
        >
          EUR
        </span>
      </div>
      <div
        class="possibleWinWrap"
      >
        <span />
        <span>
          0 EUR
        </span>
      </div>
    </div>
    <a
      class="all-in-button"
      href="javascript:;"
    >
      <span />
    </a>
    <div>
      <a
        class="bet-button"
        href="javascript:;"
      >
        <span />
      </a>
    </div>
  </div>
  <div
    class="default-keyboard"
  >
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         1
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         2
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         3
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         4
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         5
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         6
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         7
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         8
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         9
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         00
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         0
      </span>
    </a>
    <a
      class="btnWrap"
      href="javascript:;"
    >
      <span
        class="btn"
      >
         
        <span
          class="iconfont icon-jisuanqishanchu"
        />
      </span>
    </a>
  </div>
</div>
`;
